test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,330 @@
+        ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        const appRegistryAddress = (await deployments.get('AppRegistry')).address;
+        appRegistry = AppRegistry__factory.connect(appRegistryAddress, iexecAdmin);
+        const datasetRegistryAddress = (await deployments.get('DatasetRegistry')).address;
+        datasetRegistry = DatasetRegistry__factory.connect(datasetRegistryAddress, iexecAdmin);
+        const workerpoolRegistryAddress = (await deployments.get('WorkerpoolRegistry')).address;
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            workerpoolRegistryAddress,
+            iexecAdmin,
+        );
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+    }
+
+    describe('Registry', () => {
+        it('Should retrieve base URI', async () => {
+            const chainId = hre.network.config.chainId;
+            expect(await appRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/app/${chainId}/`,
+            );
+            expect(await datasetRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/dataset/${chainId}/`,
+            );
+            expect(await workerpoolRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/workerpool/${chainId}/`,
+            );
+        });
+        it('Should not set name when user is not the owner', async () => {
+            await expect(
+                appRegistry.connect(anyone).setName(ensRegistryAddress, 'new.app.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                datasetRegistry
+                    .connect(anyone)
+                    .setName(ensRegistryAddress, 'new.dataset.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                workerpoolRegistry
+                    .connect(anyone)
+                    .setName(ensRegistryAddress, 'new.workerpool.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+        it('Should not set base URI when user is not the owner', async () => {
+            await expect(
+                appRegistry.connect(anyone).setBaseURI(`https://new.url.iex.ec/app/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                datasetRegistry.connect(anyone).setBaseURI(`https://new.url.iex.ec/dataset/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                workerpoolRegistry.connect(anyone).setBaseURI(`https://new.url.iex.ec/workerpool/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+        it('Should not call initialize when user is not the owner', async () => {
```

Comment:
Why not adding should initialize with fresh registries ?

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/122/commits/f3ac7c22e9d7f41aee95143c75578fdc7d17ae04)